hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/oauth2/AzureADAuthenticator.java

```diff
@@ -275,21 +287,24 @@ public UnexpectedResponseException(final int httpErrorCode,
   }
 
   private static AzureADToken getTokenCall(String authEndpoint, String body,
-      Hashtable<String, String> headers, String httpMethod) throws IOException {
-    return getTokenCall(authEndpoint, body, headers, httpMethod, false);
+      Hashtable<String, String> headers, String httpMethod,
+      ExponentialRetryPolicy tokenFetchRetryPolicy) throws IOException {
+    return getTokenCall(authEndpoint, body, headers, httpMethod, false,
+        tokenFetchRetryPolicy);
   }
 
   private static AzureADToken getTokenCall(String authEndpoint, String body,
-      Hashtable<String, String> headers, String httpMethod, boolean isMsi)
+      Hashtable<String, String> headers, String httpMethod, boolean isMsi,
+      ExponentialRetryPolicy tokenFetchRetryPolicy)
       throws IOException {
     AzureADToken token = null;
-    ExponentialRetryPolicy retryPolicy
```

Comment:
As every OAuth token provider eventually calls this method to get token, you will just need to create an instance of exponential retry right here with the configured values ? 
Is there any benefit creating and passing down exponential retry instance from each of the token provider ?

Reply:
Done

Reply:
Done